### DISCUSSION TITLE
Looking for Guidance: Exporting and Implementing PrimeNG Figma UI Kit in Code

### DISCUSSION THREAD

**Discussion by bansheewtv** [primefaces/community#3261] [Figma UI Kit]
I'm planning to purchase the PrimeNG Figma UI Kit. I'm trying to find some videos showing how to export and how the dev will implement on the code. Any Insights would be helpful. Thanks :)

**Comment by onursenture** [comment]
We are currently working on a tool that will allow you to easily convert your theme to code by exporting tokens.json through the Tokens Studio plugin in the UI Kit. We estimate it will be released for PrimeVue by the end of January. Unfortunately, we cannot provide an estimation for PrimeNG at this time. I appreciate your interest.

**Reply by PaulCardLord** [reply]
> We are currently working on a tool that will allow you to easily convert your theme to code by exporting tokens.json through the Tokens Studio plugin in the UI Kit. We estimate it will be released for PrimeVue by the end of January. Unfortunately, we cannot provide an estimation for PrimeNG at this time. I appreciate your interest.

I have the same question regarding PrimeReact Figma UI Kit. Please tell me if the tool is ready for it? If not, is there an idea when it will be ready?
Is it possible to use tokens in a project without your tool? Using styledictionary for example https://styledictionary.com/

**Reply by onursenture** [reply]
> > We are currently working on a tool that will allow you to easily convert your theme to code by exporting tokens.json through the Tokens Studio plugin in the UI Kit. We estimate it will be released for PrimeVue by the end of January. Unfortunately, we cannot provide an estimation for PrimeNG at this time. I appreciate your interest.
> 
> I have the same question regarding PrimeReact Figma UI Kit. Please tell me if the tool is ready for it? If not, is there an idea when it will be ready? Is it possible to use tokens in a project without your tool? Using styledictionary for example https://styledictionary.com/

Hello, since the new theming is not yet available in PrimeReact, this tool is being released first for Vue and Angular. The new theming and new designer for PrimeReact are planned for Q2 2025.

**Reply by maximilian-hammerl** [reply]
@onursenture, regarding this part of your answer:

> We estimate it will be released for PrimeVue by the end of January. 

Do you already have a more precise release date and/or timeline?
And is there an issue or release plan or something like that with the current status?

**Reply by onursenture** [reply]
> @onursenture, regarding this part of your answer:
> 
> > We estimate it will be released for PrimeVue by the end of January.
> 
> Do you already have a more precise release date and/or timeline? And is there an issue or release plan or something like that with the current status?

Unfortunately, we don't have a precise release date. However, we are in the final testing stages. Likely, there won't be a delay; if there is, it will be very short.

**Reply by maximilian-hammerl** [reply]
@onursenture, how/where can find this token exporter? And is there documentation how to use it?

**Reply by onursenture** [reply]
The release is approaching. Next week is a holiday in our country, and the following week it’s planned for the implementation to be completed and testing to begin. I can’t give an estimation for the release date, but I can say that it’s coming soon.

**Reply by raconnell2** [reply]
Is using the Theme Designer the only way to convert the tokens.json file created in the Figma UI Kit into a usable format for development? This doesn't sit well with me. Especially, since the Theme Designer licensing is very expensive and a couple years ago it was announced it would be open source. Looks like it won't be now.

As @onursenture mentioned, is there no way to use https://styledictionary.com/ to covert the file ourselves? The documentation in the Figma UI Kit is woefully silent on how to go from Figma to Angular.

**Reply by onursenture** [reply]
The only **official** way to generate themes using tokens.json with the Figma UI Kit is by purchasing the Theme Designer Extended. While it’s technically possible to use Style Dictionary, we offer a paid product specifically for this purpose. Unfortunately, we cannot provide a guide on achieving this through open source or free methods. I appreciate your understanding.

**Reply by raconnell2** [reply]
This is bullocks! What use is the Figma UI Kit purchase without the ability to convert the tokens in a usable format for PrimeNG? This situation was not very well thought out from the consumer perspective. Either offer the whole thing together as a single purchase or license or not at all. This feels very much like bait and switch! Not cool.

Also, as others have mentioned. This whole process sits outside of any normal development cycle of using CICD to process things. Manually, uploading the token sets to the Theme Designer doesn't makes sense from a continuous development perspective.
